cli/release/release.go

```diff
@@ -23,3 +23,27 @@ func (r *Release) ShouldExpectFederationSecret() bool {
 		r.Configuration.Global.Datacenter != r.Configuration.Global.Federation.PrimaryDatacenter &&
 		!r.Configuration.Global.Federation.CreateFederationSecret
 }
+
+// FedSecret fetches the federation secret for the release or an empty string
+// if the federation secret is not set. The federation secret could be set
+// in several places in the configuration. This method returns the first
+// secret name that is set.
+func (r *Release) FedSecret() string {
+	if s := r.Configuration.Global.TLS.CaCert.SecretName; s != "" {
+		return s
+	}
+
+	if s := r.Configuration.Global.TLS.CaKey.SecretName; s != "" {
+		return s
+	}
+
+	if s := r.Configuration.Global.Acls.ReplicationToken.SecretName; s != "" {
```

Comment:
Good job! This should find the right secret! I only have one question/comment:

In the helm chart and RFC CSL-119, it looks like acls.enabled can be false when using WAN federation is enabled. TLS and Gossip  Encryption need to be true.  

Should you check to see if r.Configuration.Global.Acls.Enabled = true before checking for r.Configuration.GlobalAcls.ReplicationToken.SecretName ?

Reply:
Is this right? Isn't the secret `RELEASE-NAME-federation` in the secondary data center?  

I know this is not gossip key, but the createFederationSecret adds gossip key,replication token, ca cert and keyinto the federation secret. The [docs](https://www.consul.io/docs/k8s/operations/gossip-encryption-key-rotation) for rotating gossip keys also show in step 5 that in secondary data centers you need get the gossip key from `consul-federation` and it uses the key of `gossipEncryptionKey`.  So I would assume that acl replication token works the same way and gets it from the same place where [it is put](https://github.com/hashicorp/consul-k8s/blob/d708a65db4875f689fb304e77a81e577923cee8b/control-plane/subcommand/create-federation-secret/command.go#L196) ?

Reply:
I was basing the work off of [these docs](https://www.consul.io/docs/k8s/installation/multi-cluster/kubernetes#secondary-cluster-s) and just looking at any place that the user would set the secret to `consul-federation`.

Reply:
In the docs, secondary datacenters are set that way because that secret name is hardcoded here: https://github.com/hashicorp/consul-k8s/blob/d708a65db4875f689fb304e77a81e577923cee8b/control-plane/subcommand/create-federation-secret/command.go#L129

So these docs are just saying "You need to set these this way because it is a hard coded name and this consul-federation secret will have keys for gossipkey, ca cert, ca key, and replication token."

I think you want to use the hardcoded `<prefix>-federation` to check if the secret exists.  Not use this fallback pattern.

Reply:
That's a great point. Thank you for finding this. I will update the method to just return the release name with `"-federation"` appended. Now that it's so simple, it doesn't _need_ to be a method, but I like having it so that if this ever changes, we can update it in just one place.